ndcube/ndcube.py

```diff
@@ -262,105 +263,111 @@ def axis_world_coords(self, *axes, edges=False):
                 else:
-                    # If the axis is dependent on another, perform
-                    # translations on all dependent axes.
-                    # Construct pixel quantities in each dimension letting
-                    # other dimensions all have 0 pixel value.
-                    # Construct orthogonal pixel index arrays for dependent axes.
-                    quantity_list = [u.Quantity(np.zeros(tuple(
-                        [_get_dimension_for_pixel(cube_dimensions[k], edges) for k in dependent_axes[i]])),
-                        unit=u.pix)] * n_dimensions
-
-                    dependent_pixel_quantities = np.meshgrid(
-                        *[_pixel_centers_or_edges(cube_dimensions[k], edges) * u.pix
-                          for k in dependent_axes[i]], indexing="ij")
-                    for k, axis in enumerate(dependent_axes[i]):
-                        quantity_list[axis] = dependent_pixel_quantities[k]
-                # Perform wcs translation
-                dependent_axes_coords = self.pixel_to_world(*quantity_list)
-                # Place world coords into output list
-                for dependent_axis in dependent_axes[i]:
-                    if dependent_axis in int_axes:
-                        # Due to error check above we know dependent
-                        # axis can appear in int_axes at most once.
-                        j = unique_data_axis(self.wcs, dependent_axis)[0]
-                        # Since the dependent_axes_coords contains reduced number of results, adjust the index
-                        axes_coords[j] = dependent_axes_coords[j]
-                        # Remove axis from list that have now been translated.
-                        axes_translated[dependent_axes[i]] = True
-
-        # Remove the None values from axes_coords
-        axes_coords = axes_coords[axes_coords != np.array(None)]
-
-        if len(axes_coords) == 1:
-            return axes_coords[0]
-        else:
-            return tuple(axes_coords)
+                    raise TypeError(f"Unrecognized axis type: {axis, type(axis)}. "
+                                    "Must be of type (numbers.Integral, str)")
+            # Use inferred world axes to extract the desired coord value
+            # and corresponding physical types.
+            world_indices = np.array(list(world_indices), dtype=int)
+            axes_coords = np.array(axes_coords)[world_indices]
+            world_axis_physical_types = tuple(np.array(world_axis_physical_types)[world_indices])
+
+        # Return in array order.
+        # First replace characters in physical types forbidden for namedtuple identifiers.
+        identifiers = []
+        for physical_type in world_axis_physical_types[::-1]:
+            identifier = physical_type.replace(":", "_")
+            identifier = identifier.replace(".", "_")
+            identifier = identifier.replace("-", "__")
```

Comment:
Why replace this one with two?

Reply:
Good question.  The idea was that ```_``` would denote an IVOA word separator.  But ```-``` is used to show a range, e.g. ```100-200nm```, not separate words.  So I thought it would be a good idea for string manipulations to not overload the meaning of ```_```.  Perhaps there is a better replacement for ```-``` than ```__```?

Reply:
not that I can think of :+1: